client/src/pages/Shop.tsx

```diff
@@ -0,0 +1,261 @@
+const SnailTitle = styled.span`
+  ${HeadingCss};
+`;
+
+const Status = styled.div`
+  background-color: ${COLORS.PURPLE_LIGHT};
+  width: 90%;
+  padding: 15px 10px;
+
+  p {
+    text-align: center;
+  }
+`;
+
+const StarsStatus = styled(Status)`
+  display: flex;
+  flex-direction: row;
+  justify-content: center;
+  gap: 0.2rem;
+
+  p {
+    font-size: 2rem;
+  }
+`;
+
+const ItemsWrapper = styled.div`
+  background-color: ${COLORS.PURPLE_LIGHT};
+  border: 3px solid ${COLORS.PURPLE_LIGHT};
+  display: flex;
+  flex-direction: column;
+  padding: 1.5rem;
+
+  width: 68rem;
+`;
+
+function Shop() {
+  useEffect(() => {
+    const loadData = async () => {
+      // TODO: load items the user already has - don't display them
+      const snailInfo = await OWServiceProvider.getSnailInfo(username);
+      setSnailName(snailInfo.name);
+      setSnailColor(snailInfo.color);
+      setSnailImage(GetSnailImg(snailColor, 3));
+      const userInfo = await OWServiceProvider.getUserInformation(username);
+      setStarBalance(userInfo.currency);
+    };
+    loadData();
+  }, []);
+
+  const auth = useAuthUser();
+  const username = auth()?.username;
+  const [snailName, setSnailName] = useState('');
+  const [snailColor, setSnailColor] = useState('');
+  const [snailImage, setSnailImage] = useState('');
+
+  const [itemName, setItemName] = useState('');
+  const [itemType, setItemType] = useState('hat'); // TODO: Get item type dynamically
+  const [starBalance, setStarBalance] = useState(0);
+
+  return (
+    <PageWrapper pageTitle="Shop">
+      <SnailCard>
+        <SnailTitle>{snailName}</SnailTitle>
+        <img
+          src={snailImage}
+          alt={snailName + ' checking out the cool things in the shop'}
+          width="300"
+          className="snail"
+        />
+        <StarsStatus>
+          <P>
+            <b>Stars:</b> {starBalance}
+          </P>
+          <BsStars size="2rem" color={COLORS.PURPLE_MID} />
+        </StarsStatus>
+        <Status>
+          <P>
+            <b>{snailName}</b> is checking out{' '}
+            {itemType === 'glasses' ? 'these' : 'this'} cool new {itemName}{' '}
+            {itemType}. They wish they could wear{' '}
+            {itemType === 'glasses' ? 'them' : 'it'} forever!
+          </P>
+        </Status>
+        {
+          // TODO: Buy item, then go to snail page
+        }
+        <LargeRoundedButton>Buy Item</LargeRoundedButton>
+      </SnailCard>
+      <ShopCard>
+        <H1>Shop</H1>
+        <ItemsWrapper>
+          <ItemSection>
+            <H2>Snail Colors</H2>
+            <RadioWrapper>
+              <ItemSelectCard
+                item="unselected-color-1"
+                name="accessory"
+                itemResult={itemName}
+                itemType="color"
+                changeResult={setItemName}
+                changeItemType={setItemType}
+              />
+              <ItemSelectCard
+                item="unselected-color-2"
```

Comment:
These item names are ugly right now, but eventually they'll be the default colors the user didn't choose